solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1609,12 +1586,40 @@ public void publish(final CoreDescriptor cd, final Replica.State state, boolean
       if (updateLastState) {
         cd.getCloudDescriptor().setLastPublished(state);
       }
-      overseerJobQueue.offer(Utils.toJSON(m));
+      DocCollection coll = zkStateReader.getCollection(collection);
+      if (forcePublish || sendToOverseer(coll, coreNodeName)) {
```

Comment:
other places we do additional checks, like https://github.com/apache/lucene-solr/pull/2187/files#diff-5307cc9f51d88f5a171591d2f429779e22b0168cd2114275d745bccea9d1a6b3R186

please be consistent about whether we care to optimize for early termination (I suspect we don't)